### PR TITLE
Time travelling

### LINKED ISSUE #1640: WASM builds failing due to stdweb v0.4.20
### I did this

`cd server/web_ui && ./build_wasm.sh`

### I expected the following

It to build

### Kanidm version (and git commit)

Current commit, not sure which one broke it, but was recent.

### Any other comments

See https://github.com/rustwasm/wasm-bindgen/issues/3415 for details, looks like `stdweb` isn't maintained anymore, and was removed from `time` - which pulls it in - in 0.3.0.

So, we need to update ... something. I originally said `time` but that doesn't work. 😄

### PR DESCRIPTION
**Pull request by yaleman**
Fixes #1640

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run (one thing that's unrelated to this change)
- [x] cargo test has been run and passes

### CODE REVIEW

**Comment by yaleman** [comment]
This obviously can't work until v0.2.0 of `scim` is released 😄

**Comment by Firstyear** [comment]
Doing it now :)

File: Cargo.toml [hunk 1]
```diff
@@ -134,9 +134,7 @@ tide = "^0.16.0"
 # Including brotli *very* slow, so don't do that. Including the "default" feature pulls a mime-type list from the internet on build, which isn't used.
 tide-compress = { version="0.10.6", default-features = false, features = [ "gzip", "regex-check" ] }
 tide-openssl = "^0.1.1"
-# Unable to increase version due to removing ability to detect
-# local platform time.
-time = "=0.2.27"
+time = { version = "^0.3.0", features = ["formatting", "local-offset"] }
```
Comment (Firstyear) [inline Cargo.toml:137]: Needs to be 0.3.19 else a hidden cfg param is needed

File: server/core/src/actors/v1_read.rs [hunk 2]
```diff
@@ -199,8 +199,16 @@ impl QueryServerReadV1 {
         trace!(eventid = ?msg.eventid, "Begin online backup event");
 
         #[allow(deprecated)]
```
Comment (Firstyear) [inline server/core/src/actors/v1_read.rs:201]: Should we remove the allow deprecated?

File: server/lib/src/valueset/session.rs [hunk 3]
```diff
@@ -360,11 +378,16 @@ impl ValueSetT for ValueSetSession {
...
+                        #[allow(clippy::expect_used)]
+                        odt.format(&time::format_description::well_known::Rfc3339)
+                            .expect("Failed to format timestamp into RFC3339!")
                     }),
                     issued_at: {
                         debug_assert!(m.issued_at.offset() == time::UtcOffset::UTC);
-                        m.issued_at.format(time::Format::Rfc3339)
+                        #[allow(clippy::expect_used)]
+                        m.issued_at
+                            .format(&time::format_description::well_known::Rfc3339)
```
Comment (Firstyear) [inline server/lib/src/valueset/session.rs:389]: Should we just make Rfc3339 a "pub use" in lib.rs preludo to save using it like this so much?

File: Cargo.toml (hunk 1 above)
Comment (yaleman) [reply Cargo.toml:137]: Nice of them to be incredibly vague about this in the changelogs :rage:

Done!

File: server/core/src/actors/v1_read.rs (hunk 2 above)
Comment (yaleman) [reply server/core/src/actors/v1_read.rs:201]: Done

File: server/lib/src/valueset/session.rs (hunk 3 above)
Comment (yaleman) [reply server/lib/src/valueset/session.rs:389]: Done, with use statements elsewhere.

**Review by Firstyear — APPROVED** [review]

**Pull request merged**